wormhole-connect/src/routes/sdkv2/route.ts

```diff
@@ -277,13 +206,33 @@ export class SDKv2Route {
     const toChain = await this.getV2ChainContext(toChainV1);
     const sourceTokenV2 = config.sdkConverter.toTokenIdV2(sourceToken);
 
-    return (
-      await this.rc.supportedDestinationTokens(
-        sourceTokenV2,
-        fromChain.context,
-        toChain.context,
-      )
-    )
+    const destTokenIds = await this.rc.supportedDestinationTokens(
+      sourceTokenV2,
+      fromChain.context,
+      toChain.context,
+    );
+
+    // TODO SDKV2 hack
+    //
+    // SDKv2 only returns token addresses, not metadata like names and logos.
+    //
+    // For custom tokens with no built-in foreign asset addresses, this means
+    // we can't match the result of supportedDestinationTokens back up to a TokenConfig
+    // to get its name and logo.
+    //
+    // Since token bridge only outputs the same token as you put in, we're just
+    // returning sourceToken here as a hack so that we can maintain the name and logo.
+    //
+    // A longer term solution to this might be to add methods to SDKv2 for fetching token
+    // metadata like name/logo and not relying on configuration for this at all. At that
+    // point all that would be required would be an address.
+    if (this.TYPE === 'bridge' || this.TYPE === 'relay') {
```

Comment:
We should also add cosmosGateway as it uses wrapped assets

Reply:
to be honest, I want to get rid of `TYPE` completely. we have to do that, to support plugin routes, which Mayan is supposed to be. but I can do this temporarily.